jenkins/github_helpers.py

```diff
@@ -5,205 +5,199 @@
-            "Unable to read file: {}".format(file_path)
+            "Could not connect to the repository: {}. "
+            "Please make sure you are using the correct "
+            "credentials and try again.".format(repo_name)
         )
-    return data
-
-
-def update_list_of_files(repository, repo_root, file_path_list, commit_message, sha, username):
-    input_trees_list = []
-    base_git_tree = repository.get_git_tree(sha)
-    for file_path in file_path_list:
-        content = get_file_contents(repo_root, file_path)
-        input_tree = InputGitTreeElement(file_path, "100644", "blob", content=content)
-        input_trees_list.append(input_tree)
-    if len(input_trees_list) > 0:
-        new_git_tree = repository.create_git_tree(input_trees_list, base_tree=base_git_tree)
-        parents = [repository.get_git_commit(sha)]
-        author = InputGitAuthor(username, get_user_email())
-        commit_sha = repository.create_git_commit(
-            commit_message, new_git_tree, parents, author=author, committer=author
-        ).sha
-        return commit_sha
+
+    def branch_exists(self, repository, branch_name):
+        """
+        Checks to see if this branch name already exists
+        """
+        try:
+            repository.get_branch(branch_name)
+        except:
+            return False
+        return True
+
+    def get_modified_files_list(self, repo_root):
+        """
+        Use the Git library to run the ls-files command to find
+        the list of files modified.
+        """
+        git_instance = Git(repo_root)
+        git_instance.init()
+        modified_files = git_instance.ls_files("--modified")
+
+        if len(modified_files) > 0:
+            return modified_files.split("\n")
+        else:
+            return []
+
+    def create_branch(self, repository, branch_name, sha):
+        """
+        Create a new branch with the given sha as its head.
+        """
+        try:
+            branch_object = repository.create_git_ref(branch_name, sha)
+        except:
+            raise Exception(
+                "Unable to create git branch: {}. "
+                "Check to make sure this branch doesn't already exist.".format(branch_name)
+            )
+        return branch_object
+
+    def is_requirements_pr(self, branch_name):
+        return re.match("(jenkins/{0})-[a-zA-Z0-9]*".format(UPGRADE_JOB_BRANCH_NAME), branch_name)
+
+    def close_existing_pull_requests(self, repository, user_login, user_name):
+        """
+        Close any existing PR's by the bot user in this PR. This will help
+        reduce clutter, since any old PR's will be obsolete.
+        """
+        pulls = repository.get_pulls(state="open")
+        deleted_pull_numbers = []
+        for pr in pulls:
+            user = pr.user
+            if user.login == user_login and user.name == user_name:
+                branch_name = pr.head.ref
+                if not self.is_requirements_pr(branch_name):
```

Comment:
@jmbowman I've added this check here to only close old PRs for upgrade requirements, (now this PR is waiting for your review)